webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -789,75 +862,91 @@ class PolicyWriterServiceImpl(
+   * @param nodeFolder
    * @param backupFolder
    */
-  private[this] def backupNodeFolder(nodeFolder: String, backupFolder: String): Unit = {
-    val src = new File(nodeFolder)
-    if (src.isDirectory()) {
-      val dest = new File(backupFolder)
-      if (dest.isDirectory) {
-        // force deletion of previous backuup
-        FileUtils.deleteDirectory(dest)
+  private[this] def backupNodeFolder(nodeFolder: String, backupFolder: String): IOResult[Unit] = {
+    IOResult.effect {
+      val src = File(nodeFolder)
+      if (src.isDirectory()) {
+        val dest = File(backupFolder)
+        if (dest.isDirectory) {
+          // force deletion of previous backuup
+          dest.delete(false, File.LinkOptions.noFollow)
+        }
+        PolicyLogger.trace(s"Backup old '${nodeFolder}' into ${backupFolder}")
+        dest.parent.createDirectoryIfNotExists(true)
+        src.moveTo(dest)(File.CopyOptions.atomically)
       }
-      FileUtils.moveDirectory(src, dest)
     }
   }
 
   /**
    * Move the newly created folder to the final location
-   * @param newFolder : where the promises have been written
-   * @param nodeFolder : where the promises will be
+   * @param sourceFolder : where the promises have been written
+   * @param destinationFolder : where the promises will be
    */
-  private[this] def moveNewNodeFolder(sourceFolder: String, destinationFolder: String): Unit = {
-    val src = new File(sourceFolder)
-
-    logger.trace("Moving folders from %s to %s".format(src, destinationFolder))
+  private[this] def moveNewNodeFolder(sourceFolder: String, destinationFolder: String): IOResult[Unit] = {
 
-    if (src.isDirectory()) {
-      val dest = new File(destinationFolder)
+    val src = File(sourceFolder)
 
-      if (dest.isDirectory) {
-        // force deletion of previous promises
-        FileUtils.deleteDirectory(dest)
-      }
-      FileUtils.moveDirectory(src, dest)
-
-      // force deletion of dandling new promise folder
-      if ( (src.getParentFile().isDirectory) && (src.getParent().endsWith("rules.new"))) {
-        FileUtils.deleteDirectory(src.getParentFile())
-      }
-    } else {
-      logger.error("Could not find freshly created policies at %s".format(sourceFolder))
-      throw new IOException("Created policies not found !!!!")
-    }
+    for {
+      b <- IOResult.effect(src.isDirectory)
+      _ <- if (b) {
+             val dest = File(destinationFolder)
+             for {
+               _ <- PolicyLoggerPure.trace(s"Moving folders: \n  from ${src.pathAsString}\n  to   ${dest.pathAsString}")
+               _ <- ZIO.whenM(IOResult.effect(dest.isDirectory)) {
+                      // force deletion of previous promises
+                      IOResult.effect(dest.delete(false, File.LinkOptions.noFollow))
+                    }
+               _ <- IOResult.effect {
+                      dest.parent.createDirectoryIfNotExists(true)
+                      src.moveTo(dest)(File.CopyOptions.default)
```

Comment:
this ought to be atomically, there is no reason for the backup to be and not this one

Reply:
it's not all consistant

Reply:
:+1: